### PR TITLE
nginx: update to version 1.15.10

### PR DESCRIPTION
**Pull request by Ansuel**
@hnyman 

Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>

### CODE REVIEW

**Comment by diizzyy** [comment]
We should really fix the ssl dep by now...

**Comment by Ansuel** [comment]
how ? should we add a dep for openssl engine support ?

**Comment by diizzyy** [comment]
No, default (non ssl) pulls in ssl
https://github.com/openwrt/packages/issues/8125
(No, I haven't gotten around to test those modules yet) :-/

**Comment by Ansuel** [comment]
i thought you were talking about nginx broken with openssl without engine support

**Comment by Ansuel** [comment]
@diizzyy think we should test this but should be ok... it the Documentation are true

**Comment by cotequeiroz** [comment]
@Ansuel 
I am trying to reproduce the problem you reported, that SSL requires ENGINE support.  I am not able to.  Are you sure it was not an openssl build left over somewhere in your tree?  I've seen problems like this when you link to a library with engine support, and then try to run it with a library without it.  When I build it here, I have no problems, and saw no ENGINE symbols in the resulting binary.

**Comment by Ansuel** [comment]
could be that i didn't clean nginx when i recompiled openssl ( i repoted this to you as another user had this problem on a custom build on the forum) will report 

Sorry about the mess...

**Comment by cotequeiroz** [comment]
Yes, that would create the problem you described, as nginx would pick up engine code at build time, and then can't find it at run-time.  I will try once again, from scratch.

**Comment by Ansuel** [comment]
@cotequeiroz if you want to try... compile nginx with openssl with engine support. then recompile openssl without engine support and test

**Comment by cotequeiroz** [comment]
I know for sure that will cause trouble, but it is expected behavior.  You can't build it with full support and then try to run it in a lesser environment.  We should not add the dependency here for these cases.  What you could do is add `CONFIG_OPENSSL_ENGINE` to nginx's `PKG_COFNIG_DEPENDS`, so nginx is rebuild when that config changes.

**Comment by Ansuel** [comment]
@cotequeiroz should be good this way right?

File: net/nginx/Makefile [hunk 1]
```diff
@@ -71,6 +71,7 @@ PKG_CONFIG_DEPENDS := \
 	CONFIG_NGINX_STREAM_SSL_PREREAD_MODULE \
 	CONFIG_NGINX_RTMP_MODULE \
 	CONFIG_NGINX_TS_MODULE \
+	CONFIG_OPENSSL_ENGINE \
```
Comment (cotequeiroz) [inline net/nginx/Makefile]: This is what I meant.  Notice that it comes with a side effect: nginx will be rebuilt if the `OPENSSL_ENGINE` option is changed even if SSL is turned off.  Considering the fact that SSL should _almost_ always be used, I'd leave it here like this.

Comment (Ansuel) [reply net/nginx/Makefile]: IMHO not a problem.

Comment (Ansuel) [reply net/nginx/Makefile]: Typo fixed

**Comment by neheb** [comment]
Looks good to me as well. Merging.

**Comment by hnyman** [comment]
Note to the nginx-ssl-engine discussion:

I believe that the problem mentioned on the forum arises when somebody uses opkg to install and then uses the buildbot-compiled nginx to a private system. I think that the buildbot currently builds openssl with non-default engine options (as engine is forced on by some packages like libp11) so nginx possibly picks that support in the compile phase in buildbot, while a private firmware built with the default openssl settings has no engine support. Thus the buildbot-built nginx with engine-support does nto work with the normal self-built openssl.  (Buildbot builds all packages, so a config symbol forced by another package with +@ may thus also affect third packages.)

@cotequeiroz 
> Yes, that would create the problem you described, as nginx would pick up engine code at build time, and then can't find it at run-time. I will try once again, from scratch.

You might test with buildbot-like conditions: first build openssl and nginx normally to get your own baseline, then use menuconfig to add libp11 to the build and then rebuild openssl and nginx. Results probably differ from the original baseline and now the buildbot-like new nginx package likely expects engine. 

I have not had time to test this engine config options hassle more closely, but wanted to mention the thoughts I have about the possible root cause. I documented my thoughts earlier at https://github.com/openwrt/packages/pull/8341#issuecomment-478322454

**Comment by cotequeiroz** [comment]
I've done this before and causes the problem every time.  You don't necessarily need to use engine code, as libssl will make the engine calls, and you may end up with the undefined symbols when using with a leaner openssl.
The engine problem never showed up before because the option given to turn it off in 1.0.2 was misspelled and no one (not even the configure script) noticed it.  It was always compiled.  So one solution would be to return to previous behavior and always build it, or to set it by default and add a warning to the help text about not being able to install some packages from the standard repo.
It is not just the engine option.  For example, if you build openssl with NPN, nginx will pick up their symbols just like it did with engines.  Thankfully, no package has selected it so far.  NPN has been deprecated and should be gone by now.  I'll see about removing it.
We may need to do that with other non-default options being turned on somewhere else.  I haven't checked what options are being forced, but I already know ECC is forced by unbound.  It already is a default option, so it doesn't suffer from this.

**Comment by neheb** [comment]
My suggestion is to still remove the offending packages (libp11) from the tree. ping @dangowrt

**Comment by hnyman** [comment]
> My suggestion is to still remove the offending packages (libp11) from the tree

An alternative fix could be to revert the recent additition of +@ by https://github.com/openwrt/packages/commit/da07a9eb7cb58501275ed18fc5e9a19af7047d4d and go back to a simple @ config dependency, so that libp11 could only be compiled when the engine support has first been toggled on. Then the buildbot could not compile it, but that would still be more mild than the alternative proposal of  removing the package wholly.

**Comment by neheb** [comment]
Right, but it still throws up a warning when make is ran (even when not selected).

**Comment by jow-** [comment]
The proper solution would be to get rid of all `+@...` dependencies tree-wide. They lead to these kind of consistency issues over and over again and mixing `+@` with `@` will cause recursive dependency chains, triggering workaround all over the feed.

**Comment by cotequeiroz** [comment]
I was giving this some thought, because I think it is not restricted to `@` symbols, packages can do the same thing.  If a package has a `@` dependency, then anyone selecting it with a `+` has to depend on those symbols, otherwise you end up selecting a symbol without the dependent symbol being set (I'm not sure I even can understand my poor explanation).  Anyway, I came up with an idea for the python circular dependencies that keep popping up, which I'm describing at #8535, where I think is right on the subject.  If you look at the tree I show there, it may be easier to understand.

**Pull request merged**